### DISCUSSION TITLE
rmcell -> seg fault

### DISCUSSION THREAD

**Discussion by wverestek** [neperfepx/neper#436] [Q&A] [answered]
Dear all, 

I am starting from custom seeds and finally get to an overall structure that i am happy with. But, there remain some very small grains which i wanted to delete with -transform "rmcell(...),grow,renumber", which leads to a segfault.

In this discussion https://github.com/rquey/neper/discussions/339 I read that there is/was a bug related to rmcell. Is this still active?
Is there an older version where this bug was not yet introduced that I could use? Or is there a different path with a raster file to filter out those small grains? Is it possible to go back from raster to tess?

Regards
Wolfgang

**Comment by rquey** [comment, marked as answer]
I understand that you would like to generate a tess file; however, `grow,renumber` is only for a tesr.  All you can do for a tess is to proceed in successive steps:

Example:
```
# Generate nominal tessellation, export cell sizes, seed coos and weights, and cell orientations
neper -T -n 100 -dim 2 -morpho gg  -statcell size -statseed x,y,w -for tess,ori -o n100

# Exclude 10 smaller cells, write values to tmp file 
paste n100.stcell n100.stseed n100.ori | sort -nk1 | tail -90 > tmp

# Write seed coos, seed weights and cell orientations to file
awk '{print $2,$3}' tmp > coo
awk '{print $4}' tmp > weight
awk '{print $5,$6,$7}' tmp > ori

# Generate new tess (with 10 less grains) from new seed coos and weights and cell orientations
neper -T -n 90 -dim 2 -morphooptiini "coo:file(coo),weight:file(weight)" -ori "file(ori)" -o n90

neper -V n100.tess -datacellcol ori -print n100
neper -V n90.tess  -datacellcol ori -print n90
```

![image](https://user-images.githubusercontent.com/2659376/162736898-75d6225f-2883-498a-a5fa-c373816f1771.png)
![image](https://user-images.githubusercontent.com/2659376/162736978-56cdcb38-c19f-4249-9f12-d3b814685c4e.png)

**Reply by wverestek** [reply]
Hi,

nice workaround. I didn't think of that route. 
Thanks
